### PR TITLE
docs: better openapi spec

### PR DESCRIPTION
**Pull request by dabeeeenster**
- Moved docusaurus to Typescript
- Move the API auto-generation to https://github.com/PaloAltoNetworks/docusaurus-openapi-docs

### CODE REVIEW

File: docs/static/openapi/edge-api.yaml [hunk 1]
```diff
@@ -0,0 +1,225 @@
+openapi: 3.0.3
```
Comment (kyle-ssg) [inline docs/static/openapi/edge-api.yaml]: Do we need a note saying this is for interaction with our SDK endpoints / link to admin API swagger for other functions? I think as a first time user clicking API from docs it looks like these are the only apis we offer.

Comment (dabeeeenster) [reply docs/static/openapi/edge-api.yaml]: I've added an overview page that makes things clearer here

File: docs/static/openapi/edge-api.yaml [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+          type: array
+          $ref: '#/components/schemas/FeatureState'
+    FeatureState:
+      type: object
+      properties:
+        enabled:
+          type: boolean
+        feature_state_value:
+          type: Can be any of the following - string, integer, boolean, float
+          example: 'a-remote-config-value'
```
Comment (khvn26) [inline docs/static/openapi/edge-api.yaml]: Strictly, this should be a [`oneOf`](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/).

File: docs/static/openapi/edge-api.yaml [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      tags: ['Edge API']
+      summary: Get flags
+      description: Retrieve the flags for an environment.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Flags'
```
Comment (khvn26) [inline docs/static/openapi/edge-api.yaml]: I'd love this to be the reality but unfortunately this schema is not correct.

Currently the `/flags` endpoint returns either:
- an array of flag objects
- a single flag object — if `feature` query parameter provided to the endpoint

**Review by khvn26 — CHANGES_REQUESTED** [review]
Sorry for the long wait — dropped a couple comments.

File: .github/workflows/docs-pull-request.yml [hunk 4]
```diff
@@ -29,8 +29,6 @@ jobs:
 
       - name: npm install
         run: npm ci --include=dev --ignore-scripts
+
       - name: Run docusaurus build to check for broken links etc
-        run: |
-          npx prettier --version
-          npx prettier --check docs
```
Comment (matthewelwell) [inline .github/workflows/docs-pull-request.yml:35]: This is being done by pre-commit now

File: docs/static/openapi/edge-api.yaml (hunk 2 above)
Comment (matthewelwell) [reply docs/static/openapi/edge-api.yaml]: Fixed.

File: docs/static/openapi/edge-api.yaml (hunk 3 above)
Comment (matthewelwell) [reply docs/static/openapi/edge-api.yaml]: Fixed (unfortunately).

**Review by khvn26 — APPROVED** [review]

File: docs/docs/edge-api/edge-api.info.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    as={"h2"}
+    className={"openapi-tabs__heading"}
+    children={"Authentication"}
+  >
+  </Heading><SchemaTabs
+    className={"openapi-tabs__security-schemes"}
+  >
+    <TabItem
+      label={"API Key: EnvironmentKeyAuth"}
+      value={"EnvironmentKeyAuth"}
```
Comment (khvn26) [inline docs/docs/edge-api/edge-api.info.mdx]: Judging from site content we should change "EnvironmentKeyAuth" to a human-readable value, but I don't know exactly how to do it.

Comment (matthewelwell) [reply docs/docs/edge-api/edge-api.info.mdx]: From what I can tell, there's no real option to do this in the OpenAPI specification. See [here](https://swagger.io/docs/specification/v3_0/authentication/api-keys/).

The best we can do perhaps, is to just drop the `Auth` from the end? Like this: 

image

What do you think?

Comment (khvn26) [reply docs/docs/edge-api/edge-api.info.mdx]: That looks good, let's do it.

Comment (matthewelwell) [reply docs/docs/edge-api/edge-api.info.mdx]: Done

**Pull request merged**